lib/generators/templates/rpush.rb

```diff
@@ -80,6 +80,23 @@
   # on.tcp_connection_lost do |app, error|
   # end
 
+  # Called for each recipient which successfully receives a notification. This
+  # can occur more than once for the same notification when there are multiple
+  # recipients.
+  # on.fcm_delivered_to_recipient do |notification|
+  # end
+
+  # Called for each recipient which fails to receive a notification. This
+  # can occur more than once for the same notification when there are multiple
+  # recipients. (do not handle invalid registration IDs here)
+  # on.fcm_failed_to_recipient do |notification, error|
+  # end
+
+  # Called when the FCM returns a failure that indicates an invalid device token.
+  # You will need to delete the device token from your records.
+  # on.fcm_invalid_device_token do |app, error, device_token|
+  # end
```

Comment:
I think the new `fcm_canonical_id` reflection is missing from the updated template?

Reply:
https://github.com/lufia/til/blob/master/fcm.rst#canonical-registration-id
I understand that Canonical ID is no longer used
(Someone please point out if I am wrong).